funnel/models/post.py

```diff
@@ -0,0 +1,260 @@
+from baseframe import __
+from coaster.sqlalchemy import StateManager, with_roles
+from coaster.utils import LabeledEnum
+
+from . import (
+    BaseMixin,
+    Commentset,
+    MarkdownColumn,
+    Profile,
+    Project,
+    TimestampMixin,
+    TSVectorType,
+    User,
+    UuidMixin,
+    Voteset,
+    db,
+)
+from .commentvote import SET_TYPE
+
+__all__ = ['Post']
+
+
+class POST_STATE(LabeledEnum):  # NOQA: N801
+    DRAFT = (0, 'draft', __("Draft"))
+    PUBLISHED = (1, 'published', __("Published"))
+    DELETED = (2, 'deleted', __("Deleted"))
+
+
+class VISIBILITY_STATE(LabeledEnum):  # NOQA: N801
+    PUBLIC = (0, 'public', __("Public"))
+    RESTRICTED = (1, 'restricted', __("Restricted"))
+
+
+class Post(UuidMixin, BaseMixin, TimestampMixin, db.Model):
+    __tablename__ = 'post'
+
+    _visibility_state = db.Column(
+        'visibility_state',
+        db.SmallInteger,
+        StateManager.check_constraint('visibility_state', VISIBILITY_STATE),
+        default=VISIBILITY_STATE.PUBLIC,
+        nullable=False,
+        index=True,
+    )
+    visibility_state = StateManager(
+        '_visibility_state', VISIBILITY_STATE, doc="Visibility state"
+    )
+
+    _state = db.Column(
+        'state',
+        db.Integer,
+        StateManager.check_constraint('state', POST_STATE),
+        default=POST_STATE.DRAFT,
+        nullable=False,
+        index=True,
+    )
+    state = StateManager('_state', POST_STATE, doc="Post state")
+
+    user_id = db.Column(None, db.ForeignKey('user.id'), nullable=False, index=True)
+    user = with_roles(
+        db.relationship(
+            User, backref=db.backref('posts', lazy='dynamic'), foreign_keys=[user_id],
+        ),
+        grants={'author'},
```

Comment:
Would `creator` be more appropriate? It is possible to be copy-pasting content, or to create a stub for someone else to fill in. This happens with proposals.